File: Exemplos/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_Flutter-font/lib/IntentDigitalHubService/intent_digital_hub_module.dart
// ignore: constant_identifier_names
enum IntentDigitalHubModule { BRIDGE, TERMICA, SAT }

extension DigitalHubExtension on IntentDigitalHubModule {
  String get intentPath {
    switch (this) {
      case IntentDigitalHubModule.BRIDGE:
        return 'com.elgin.e1.digitalhub.BRIDGE';
      case IntentDigitalHubModule.TERMICA:
        return 'com.elgin.e1.digitalhub.TERMICA';
      case IntentDigitalHubModule.SAT:
        return 'com.elgin.e1.digitalhub.SAT';
    }
  }
}

File: Exemplos M10/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_Kotlin-font/app/src/main/java/com/elgin/intent_digital_hub/IntentDigitalHubService/IntentDigitalHubModule.kt
package com.elgin.intent_digital_hub.IntentDigitalHubService

/**
 * Módulo disponíveis no Intent Digital Hub
 */
enum class IntentDigitalHubModule(val intentPath: String) {
    BALANCA("com.elgin.e1.digitalhub.BALANCA"),
    BRIDGE("com.elgin.e1.digitalhub.BRIDGE"),
    SAT("com.elgin.e1.digitalhub.SAT"),
    TERMICA("com.elgin.e1.digitalhub.TERMICA");
}